pkg/api/vlabs/validate.go

```diff
@@ -509,11 +509,6 @@ func (a *Properties) validateZones() error {
 }
 
 func (a *Properties) validateLinuxProfile() error {
-	for _, publicKey := range a.LinuxProfile.SSH.PublicKeys {
```

Comment:
my only concern with removing this validation is that we also use these functions for other aks-engine commands such as scale, upgrade, etc. and the public key is still a required value for those.

Reply:
right - this is exactly why pubkey generation should happen on Load, and validation should happen afterwards. Then we can validate that there is a pubkey present before anything else tries to use it.